packages/atomic/src/components/common/carousel.spec.ts

```diff
@@ -0,0 +1,212 @@
+    await renderComponent();
+    const showMoreButton = await locators.nextButton.element();
+    await expect(showMoreButton).toHaveAttribute('part', 'next-button');
+  });
+
+  it('should have the correct part attribute for the icon', async () => {
+    await renderComponent();
+    await expect
+      .element(locators.previousButtonIcon)
+      .toHaveAttribute('part', 'previous-icon');
+    await expect
+      .element(locators.nextButtonIcon)
+      .toHaveAttribute('part', 'next-icon');
+  });
+
+  it('should call #previousPage when the previous button is clicked', async () => {
+    const previousPageMock = vi.fn();
+
+    const carouselElement = await renderComponent({
+      previousPage: previousPageMock,
+    });
+    const previousButton = carouselElement.querySelector(
+      '[part="previous-button"]'
+    ) as HTMLElement;
+    previousButton.click();
+    expect(previousPageMock).toHaveBeenCalled();
+  });
+
+  it('should call #nextPage when the next button is clicked', async () => {
+    const nextPageMock = vi.fn();
+    const carouselElement = await renderComponent({
+      nextPage: nextPageMock,
+    });
+    const nextButton = carouselElement.querySelector(
+      '[part="next-button"]'
+    ) as HTMLElement;
+    nextButton.click();
+    expect(nextPageMock).toHaveBeenCalled();
+  });
+
+  it('should not render buttons or indicators when there is only one page', async () => {
+    const carouselElement = await renderComponent({
+      numberOfPages: 1,
+    });
+    const previousButton = carouselElement.querySelector(
+      '[part="previous-button"]'
+    );
+    const nextButton = carouselElement.querySelector('[part="next-button"]');
+    const indicators = carouselElement.querySelector('[part="indicators"]');
+
+    expect(previousButton).toBeNull();
+    expect(nextButton).toBeNull();
+    expect(indicators).toBeNull();
+  });
+
+  it('should mark the correct indicator as active', async () => {
+    const numberOfPages = 3;
+    const testCases = [
+      {currentPage: 0, expectedActive: 0},
+      {currentPage: 1, expectedActive: 1},
+      {currentPage: 2, expectedActive: 2},
+    ];
+    for (const {currentPage, expectedActive} of testCases) {
+      const carouselElement = await renderComponent({
+        numberOfPages,
+        currentPage,
+      });
+      const indicatorsContainer = carouselElement.querySelector(
+        '[part="indicators"]'
+      );
+      const indicators = indicatorsContainer
+        ? Array.from(indicatorsContainer.children)
+        : [];
+      expect(indicators.length).toBe(numberOfPages);
+      for (let i = 0; i < indicators.length; i++) {
+        if (i === expectedActive) {
+          expect(indicators[i].getAttribute('part')).toContain(
+            'active-indicator'
+          );
+        } else {
+          expect(indicators[i].getAttribute('part')).not.toContain(
+            'active-indicator'
+          );
+        }
+      }
+    }
+  });
```

Comment:
I think having a loop within a loop in a test is not ideal. 
Also, the test becomes hard to read.

My suggestion would be to split it into 2 tests and use locators to shorten the test
```suggestion
  it('should mark the current indicator as active', async () => {
    const numberOfPages = 3;
    const currentPage = 1;
    await renderComponent({numberOfPages, currentPage});
    const listItem = (page: number) => locators.indicators.nth(page);

    await expect(listItem(currentPage)).toHaveAttribute(
      'part',
      'indicator active-indicator'
    );
  });

  it('should not mark other pages active', async () => {
    const numberOfPages = 3;
    const currentPage = 1;
    await renderComponent({numberOfPages, currentPage});
    const listItem = (page: number) => locators.indicators.nth(page);

    await expect(listItem(0)).not.toHaveAttribute(
      'part',
      'indicator active-indicator'
    );
    await expect(listItem(2)).not.toHaveAttribute(
      'part',
      'indicator active-indicator'
    );
  });
```